pkg/scheduler/queue/queue.go

```diff
@@ -93,6 +93,18 @@ func (q *RequestQueue) Enqueue(tenant string, path []string, req Request, maxQue
 		return errors.New("no queue found")
 	}
 
+	// Optimistically increase queue counter for tenant.
+	// We need to keep track of queue length separately because the size of the
+	// buffered channel is the same across all sub-queues which would allow
+	// enqueuing more items than there are allowed at tenant level.
```

Comment:
Instead of doing an increase to decrease inside the if. Can't we have a `get` method for the int map and use it?
So we only call increase after the if statement.